src/components/web3-connect/index.tsx

```diff
@@ -156,6 +156,16 @@ export class Container extends React.Component<Properties, State> {
     if (web3.account !== previouslyAccount) {
       this.props.setAddress(web3.account);
     }
+
+    if (
+      connectionStatus === ConnectionStatus.Connecting &&
+      previousConnectionStatus === ConnectionStatus.Connected &&
+      web3.account
+    ) {
+      localStorage.removeItem('previousConnector');
+      window.location.reload();
+      this.reconnectPreviousConnector();
```

Comment:
Pretty sure if we reload we'll never get to this line?

Reply:
yes i am pretty sure. Spend considerable time on testing. Though it could happen there is a case (or comes a case in future) where we fallback to this line, but i am good with merging this.